snapcraft/storeapi/v2/channel_map.py

```diff
@@ -0,0 +1,266 @@
+            "version": self.version,
+            "architectures": self.architectures,
+        }
+
+    def __repr__(self) -> str:
+        return f"<Revision: {self.revision!r} for version {self.version!r} and architectures {self.architectures!r}>"
+
+    def __init__(
+        self, *, revision: int, version: str, architectures: List[str]
+    ) -> None:
+        self.revision = revision
+        self.version = version
+        self.architectures = architectures
+
+
+class SnapChannel:
+    """
+    Represents a Channel from a channel List for snap.
+    """
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Optional[str]]):
+        return cls(
+            name=payload["name"],
+            track=payload["track"],
+            risk=payload["risk"],
+            branch=payload["branch"],
+            fallback=payload["fallback"],
+        )
+
+    def marshal(self) -> Dict[str, Optional[str]]:
+        return {
+            "name": self.name,
+            "track": self.track,
+            "risk": self.risk,
+            "branch": self.branch,
+            "fallback": self.fallback,
+        }
+
+    def __repr__(self) -> str:
+        return f"<SnapChannel: {self.name!r}>"
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        track: str,
+        risk: str,
+        branch: Optional[str],
+        fallback: Optional[str],
+    ) -> None:
+        self.name = name
+        self.track = track
+        self.risk = risk
+        self.branch = branch
+        self.fallback = fallback
+
+
+class Snap:
+    """Represents a Snap structure from the Snap Store."""
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Any]) -> "Snap":
+        return cls(
+            name=payload["name"],
+            channels=[SnapChannel.unmarshal(sc) for sc in payload["channels"]],
+        )
+
+    def marshal(self) -> Dict[str, Any]:
+        return {"name": self.name, "channels": [sc.marshal() for sc in self.channels]}
+
+    def __repr__(self) -> str:
+        return "<Snap: {!r}>".format(self.name)
+
+    def __init__(self, *, name: str, channels: List[SnapChannel]) -> None:
+        self.name = name
+        self.channels = channels
+
+
+class ChannelMap:
+    """
+    Represents the data returned from the channel-map call from the Snap Store.
+    """
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Any]) -> "ChannelMap":
+        return cls(
+            channel_map=[MappedChannel.unmarshal(c) for c in payload["channel-map"]],
+            revisions=[Revision.unmarshal(r) for r in payload["revisions"]],
+            snap=Snap.unmarshal(payload["snap"]),
+        )
+
+    def marshal(self) -> Dict[str, Any]:
+        return {
+            "channel-map": [c.marshal() for c in self.channel_map],
+            "revisions": [r.marshal() for r in self.revisions],
+            "snap": self.snap.marshal(),
+        }
+
+    def __repr__(self) -> str:
+        return "<ChannelMap: {!r}>".format(self.snap.name)
```

Comment:
could use `self.__class__.__name__` in case you ever change the name

Reply:
done